six/modules/c++/six.sicd/tests/test_mesh_polyfit.cpp

```diff
@@ -0,0 +1,460 @@
+/* =========================================================================
+ * This file is part of six-c++
+ * =========================================================================
+ * 
+ * (C) Copyright 2004 - 2016, MDA Information Systems LLC
+ *
+ * six-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public 
+ * License along with this program; If not, 
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <cstdlib>
+#include <ctime>
+#include <memory>
+#include <iostream>
+#include <iomanip>
+
+#include <six/sicd/SICDMesh.h>
+#include <six/sicd/Utilities.h>
+
+namespace
+{
+void populatePlanarCoordinateMeshVectors(const types::RowCol<size_t>& meshDims,
+                                         double xMin, double xMax,
+                                         double yMin, double yMax,
+                                         std::vector<double>& x,
+                                         std::vector<double>& y)
+{
+    x.clear();
+    y.clear();
+
+    double xDelta = (xMax - xMin) / ((double) (meshDims.row - 1));
+    double yDelta = (yMax - yMin) / ((double) (meshDims.col - 1));
+    double xv = xMin;
+    for (size_t ri = 0; ri < meshDims.row; ++ri, xv += xDelta)
+    {
+        double yv = yMin;
+        for (size_t ci = 0; ci < meshDims.col; ++ci, yv += yDelta)
+        {
+            x.push_back(xv);
+            y.push_back(yv);
+        }
+    }
+}
+}
+
+int main(int argc, char** argv)
+{
+    const std::string RTNID = "[test_mesh_polyfit] ";
+    
+    double epsilon = 1e-9;
+    double xMax =  10000.0;
+    double yMax =  20000.0;
+    size_t orderX = 4;
+    size_t orderY = 4;
+    types::RowCol<size_t> slantMeshDims(7, 9);
+    types::RowCol<double> slantSampleSpacing(1.0, 1.0);
+    types::RowCol<double> slantCenter(0.0, 0.0);
+    types::RowCol<size_t> outputMeshDims(7, 9);
+    types::RowCol<double> outputSampleSpacing(1.0, 1.0);
+    types::RowCol<double> outputCenter(0.0, 0.0);
+    bool printPolys = false;
+
+    for (int ii = 1; ii < argc; ++ii)
+    {
+        std::string arg(argv[ii]);
+        if (arg == "-slant-rows")
+        {
+            if (ii == argc - 1)
+            {
+                std::cout << RTNID << "Error: -slant-rows requires a value"
+                    << std::endl;
+                return 1;
+            }
+
+            ++ii;
+            slantMeshDims.row = std::atoi(argv[ii]);
```

Comment:
If you keep it like this rather than cli, these should all be `str::toType` instead of `std::atoi`